From 8fb6a45a738520cae83070ae22c9d4652d8c09c9 Mon Sep 17 00:00:00 2001
From: Gerald Pfeifer <gerald@pfeifer.com>
Date: Sun, 6 Apr 2025 23:19:45 +0200
Subject: [PATCH 359/441] ntdll: Fix build on platforms without getauxval.

Commit b1b5105ceb moved the following block

    #ifdef AT_HWCAP2
             if (getauxval( AT_HWCAP2 ) & 2) syscall_flags |= SYSCALL_HAVE_WRFSGSBASE;
    #endif

out of an #ifdef __linux__ block and it now causes failures on FreeBSD
and others which do not have getauxval.
---
 dlls/ntdll/unix/signal_x86_64.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/ntdll/unix/signal_x86_64.c b/dlls/ntdll/unix/signal_x86_64.c
index 538f6dcf0c1..497946b67ab 100644
--- a/dlls/ntdll/unix/signal_x86_64.c
+++ b/dlls/ntdll/unix/signal_x86_64.c
@@ -2979,11 +2979,11 @@ void signal_init_process(void)
 
     if (cpu_info.ProcessorFeatureBits & CPU_FEATURE_XSAVE) syscall_flags |= SYSCALL_HAVE_XSAVE;
     if (xstate_compaction_enabled) syscall_flags |= SYSCALL_HAVE_XSAVEC;
+
+#ifdef __linux__
 #ifdef AT_HWCAP2
     if (getauxval( AT_HWCAP2 ) & 2) syscall_flags |= SYSCALL_HAVE_WRFSGSBASE;
 #endif
-
-#ifdef __linux__
     if (wow_teb)
     {
         int sel;
-- 
2.51.0

